src/modules/RootPart.rb

```diff
@@ -2040,22 +2038,21 @@ def FindRootPartitions
       @numberOfValidRootPartitions = 0
 
       # all formatted partitions and lvs on all devices
-      filesystems = probed.filesystems.with { |fs| fs.to_s != "swap" }
-      partitions = filesystems.partitions.to_a + filesystems.lvm_lvs.to_a
+      filesystems = probed.blk_filesystems.select { |fs| fs.type.to_sym != :swap }
```

Comment:
NP: Just a suggestion `probed.blk_filesystems.reject { |fs| fs.type.is?(:swap) }`